lms/djangoapps/courseware/model_data.py

```diff
@@ -500,6 +500,11 @@ def set_score(self, user_id, usage_key, score, max_score):
             student_module.max_grade = max_score
             student_module.save()
 
+
+    @contract(username="basestring", block_key=UsageKey)
+    def get_history(self, username, block_key):
+        return self._cache.get_history(username, block_key)
```

Comment:
I think I'd lean towards not putting the history methods in the `UserStateCache` at all, and just have the places that need the history instantiate the `UserStateClient` directly.